src/main/java/org/folio/services/kafka/topic/KafkaAdminClientService.java

```diff
@@ -53,11 +53,34 @@ public KafkaAdminClientService(Supplier<KafkaAdminClient> clientFactory) {
             log.error("Failed to close kafka admin client", closeResult.cause());
           }
         });
+      });
+  }
+
+  public Future<Void> deleteKafkaTopics(String tenantId, String environmentName) {
+    Promise<Void> result = Promise.promise();
+    final KafkaAdminClient kafkaAdminClient = clientFactory.get();
+    List<String> topicsToDelete = readTopics()
+      .map(topic -> qualifyName(topic, environmentName, tenantId))
+      .map(NewTopic::getName)
+      .collect(Collectors.toList());
+    kafkaAdminClient.deleteTopics(topicsToDelete, result);
+    result.future().onComplete(res -> {
+      if (res.succeeded()) {
+        log.info("Topics deleted successfully");
+      } else {
+        log.error("Unable to delete topics", res.cause());
+      }
+      kafkaAdminClient.close().onComplete(closeResult -> {
+        if (closeResult.failed()) {
+          log.error("Failed to close kafka admin client", closeResult.cause());
+        }
+      });
     });
+    return result.future();
```

Comment:
Avoiding the Promise and using dedicated Future methods:
```
    return kafkaAdminClient.deleteTopics(topicsToDelete)
        .onSuccess(x -> log.info("Topics deleted successfully"))
        .onFailure(e -> log.error("Unable to delete topics", e))
        .eventually(x -> 
            kafkaAdminClient.close()
            .onFailure(e -> log.error("Failed to close kafka admin client", e)));
```

Creating and closing the AdminKafkaClient is duplicated code (createKafkaTopics) and can be extracted into a `withAdminKafkaClient` method:

```
  public Future<Void> deleteKafkaTopics(String tenantId, String environmentName) {
    List<String> topicsToDelete = readTopics()
      .map(topic -> qualifyName(topic, environmentName, tenantId))
      .map(NewTopic::getName)
      .collect(Collectors.toList());
    return withKafkaAdminClient(kafkaAdminClient -> kafkaAdminClient.deleteTopics(topicsToDelete))
        .onSuccess(x -> log.info("Topics deleted successfully"))
        .onFailure(e -> log.error("Unable to delete topics", e));
  }

  private <T> Future<T> withKafkaAdminClient(Function<KafkaAdminClient, Future<T>> function) {
    final KafkaAdminClient kafkaAdminClient = clientFactory.get();
    return function.apply(kafkaAdminClient)
        .eventually(x ->
            kafkaAdminClient.close()
            .onFailure(e -> log.error("Failed to close kafka admin client", e)));
  }
```